### PR TITLE
yaml support for train endpoint

### PR DESCRIPTION
**Pull request by wochinge**
**Proposed changes**:
- fix https://github.com/RasaHQ/rasa/issues/6354

**Status (please check what you already did)**:
- [x] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: rasa/server.py [hunk 1]
```diff
@@ -1115,3 +1053,128 @@ def _get_output_channel(
...
+
+def _model_output_directory(save_to_default_model_directory: bool) -> Text:
+    if save_to_default_model_directory is True:
+        return DEFAULT_MODELS_PATH
+
+    return tempfile.gettempdir()
+
+
+def _validate_yaml_training_payload(yaml_text: Text) -> None:
+    # TODO: Use NLU schema validation once it's merged
```
Comment (wochinge) [inline rasa/server.py:1170]: depends on https://github.com/RasaHQ/rasa/pull/6256

File: tests/test_server.py [hunk 2]
```diff
@@ -288,7 +290,7 @@ def is_server_ready() -> bool:
         ),
     ],
 )
-def test_parse(rasa_app, response_test):
+def test_parse(rasa_app: SanicTestClient, response_test):
```
Comment (ricwo) [inline tests/test_server.py:293]:
```suggestion
def test_parse(rasa_app: SanicTestClient, response_test: ResponseTest):
```

File: tests/test_server.py [hunk 3]
```diff
@@ -330,7 +332,9 @@ def test_parse(rasa_app, response_test):
         ),
     ],
 )
-def test_parse_with_different_emulation_mode(rasa_app, response_test):
+def test_parse_with_different_emulation_mode(
+    rasa_app: SanicTestClient, response_test: ResponseTest
```
Comment (ricwo) [inline tests/test_server.py:336]: 🚀

File: changelog/6354.feature.md [hunk 4]
```diff
@@ -0,0 +1,4 @@
+The Rasa Open Source API endpoint `POST /model/train` now supports training data in YAML
+format. Please specify the header `Content-Type: application/yaml` when
+training a model using YAML training data. 
+See the [API documentation](./api/http-api.mdx) for more information.
```
Comment (ricwo) [inline changelog/6354.feature.md:4]: Should we say a bit more about how the new request parameters affect the `force` and `default_directory` keys in the json payload?

File: docs/static/spec/rasa.yml [hunk 5]
```diff
@@ -435,12 +435,32 @@ paths:
...
       requestBody:
         required: true
+        description: >-
+          The training data can either be in YAML format or a JSON
+          which contains Rasa training data in Markdown format for each required key.
         content:
           application/json:
             schema:
               $ref: '#/components/schemas/TrainingRequest'
+          application/yaml:
```
[the remark this answers is not in the document]
Comment (wochinge) [reply docs/static/spec/rasa.yml:460]: Ah, I looked for it but couldn't find anything. Turns out it's in the frontend code. Will change it.

File: rasa/server.py [hunk 6]
```diff
@@ -1115,3 +1053,128 @@ def _get_output_channel(
...
+        ),
+    )
+
+    return dict(
+        domain=domain_path,
+        config=config_path,
+        training_files=temp_dir,
+        output=model_output_directory,
+        force_training=request_payload.get(
+            "force", request.args.get("force_training", False)
```
[the remark this answers is not in the document]
Comment (wochinge) [reply rasa/server.py:1107]: deprecated it

File: changelog/6354.feature.md (hunk 4 above)
Comment (wochinge) [reply changelog/6354.feature.md:4]: added another changelog

Comment (ricwo) [reply changelog/6354.feature.md:4]: don't think that's checked in yet

Comment (wochinge) [reply changelog/6354.feature.md:4]: 🙈

File: changelog/6354.removal.md [hunk 7]
```diff
@@ -0,0 +1,4 @@
+Specifying the parameters `force` and `save_to_default_model_directory` as part of the
+JSON payload when training a model using `POST /model/train` is now deprecated.
+Please use the header arguments `force_training` and `save_to_default_model_directory`
```
Comment (ricwo) [inline changelog/6354.removal.md:3]: aren't these 
```suggestion
Please use the request parameters `force_training` and `save_to_default_model_directory`
```
?

File: rasa/server.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (ricwo) [reply rasa/server.py:1107]: `request.args` gets the query string variables (see comment in API spec)

File: docs/static/spec/rasa.yml [hunk 8]
```diff
@@ -435,12 +435,34 @@ paths:
         Trains a new Rasa model. Depending on the data given only a dialogue model,
         only a NLU model, or a model combining a trained dialogue model with an
         NLU model will be trained. The new model is not loaded by default.
+      parameters:
+      - in: header
```
Comment (ricwo) [inline docs/static/spec/rasa.yml:439]:
```suggestion
      - in: query
```

File: docs/static/spec/rasa.yml [hunk 9]
```diff
@@ -435,12 +435,34 @@ paths:
...
+        name: save_to_default_model_directory
+        schema:
+          type: boolean
+          default: True
+          description: >-
+            If `true` (default) the trained model will be saved in the default model
+            directory, if `false` it will be saved in a temporary directory
+      - in: header
```
Comment (ricwo) [inline docs/static/spec/rasa.yml:447]:
```suggestion
      - in: query
```

File: changelog/6354.removal.md (hunk 7 above)
Comment (wochinge) [reply changelog/6354.removal.md]: yeah, you're completely right! 🙈

**Pull request merged**